x-pack/plugins/connectors_networking/server/docs/poc.md

```diff
@@ -0,0 +1,345 @@
+
+Does not include client-side certs, keys, etc. Not clear we need this, and at
+least seems like it can be deferred.  Client-side certs/keys are needed for
+more secure networking where the server need to authenticate the client at
+the TLS level.  It's not used as much as the basic TLS ca certificates, which
+are used to let the client authenticate the server at the TLS level.
+
+### CNO's are space agnostic
+
+For our current usage, it's not clear that we would need space-specific CNO's,
+since the options are all about validating the server is who it says it is. 
+Would one space want to verify a server, and another not?
+
+I guess we have to think about the future.  What if we want to add something here
+that _should_ be space specific.  It would certainly be some new sort of data,
+so would probably require a new saved object, which _could_ be space specific.
+
+### CNO's are unique by protocol : hostname : port
+
+For now anyway, it seems possible to get by with having CNO objects be unique
+just by protocol / hostname / port - without considering any path, query string
+parameters, user/pass, etc. TLS certificates only go to this granularity as
+well. If needed, in the future, we could extend this with path's or other
+factors.
+
+We'll canonicalize the url, by only using the protocol, hostname, and port from
+provided urls, and add the default port for the protocol if not provided in the
+url.
+
+This breaks for the only generic option, `timeout`.  Presumably you should be
+able to set a specific timeout for a specific url pattern that could include
+a path.  For example, you might use the default timeout for most requests,
+but then a certain API like `/api/digits-of-pi` might need a longer timeout,
+so it would be nice to constrain the longer timeout to just that URL.  Perhaps
+that's another good reason to remove the `timeout` option.  We could re-visit
+allowing path elements (and query string params? headers?) later.
+
+### CNO's are "matched" to URLs, not specifically assigned
+
+Rather than have to assign a CNO to a specific connector, we're taking
+advantage of the fact that TLS options are specific to the hostname and port,
+and only applicable to https and smtp.  This allows us to create the CNO's
+independent of the connectors - they could be created BEFORE the connectors
+are created.  And once created, and new connectors using the same hostname
+and port will pick up the CNO without any direct association.
+
+This works out great, in practice, but am wondering if I missed some gotchas
+with it.
+
+In fact, having written that bit above about creating CNO's "early", these CNO's
+are perhaps good fodder for Kibana config - a way of pre-registering them, where
+we could (potentially) have the customers provide PEMs as file names (for
+on-prem; clould would need to support PEM's as strings of the file contents (7
+bit ascii anyway!)).
+
+In fact, having written that, if we could do this, we could perhaps ship this
```

Comment:
This certainly simplifies the implementation, but puts additional burden on the administrator. We've seen what happens with the ESO encryption key, and I suspect we'll have similar problems with yml-based configuration here.

For the record, I think a yml-based approach makes sense as a starting point, especially if we're OK with space-agnostic (global) settings. We'll just want to make our expectations clear in the docs, and provide helpful error messages when possible.

Reply:
So if we decide space-aware ESOs in the possible future, but we have "preconfigured" global options objects now, that mirrors how we do preconfigured connectors - in specified in yaml, they're in every space, but the http api only creates space-aware connectors.